pkg/trace/api/normalizer.go

```diff
@@ -21,48 +23,78 @@ const (
 	MaxNameLen = 100
 	// MaxTypeLen the maximum length a span type can have
 	MaxTypeLen = 100
-	// MaxEndDateOffset the maximum amount of time in the future we
-	// tolerate for span end dates
-	MaxEndDateOffset = 10 * time.Minute
+	// DefaultServiceName is the default name we assign a service if it's missing and we have no reasonable fallback
+	DefaultServiceName = "service"
+	// DefaultSpanName is the default name we assign a span if it's missing and we have no reasonable fallback
+	DefaultSpanName = "service.trace"
 )
 
 var (
 	// Year2000NanosecTS is an arbitrary cutoff to spot weird-looking values
 	Year2000NanosecTS = time.Date(2000, time.January, 1, 0, 0, 0, 0, time.UTC).UnixNano()
 )
 
-// normalize makes sure a Span is properly initialized and encloses the minimum required info
-func normalize(s *pb.Span) error {
+// normalize makes sure a Span is properly initialized and encloses the minimum required info, returning error if it
+// is invalid beyond repair
+func normalize(ts *info.TagStats, s *pb.Span) error {
+	fallbackServiceName := DefaultServiceName
+	if ts.Lang != "" {
+		fallbackServiceName = ts.Lang
+	}
+
+	if s.TraceID == 0 {
+		atomic.AddInt64(&ts.TracesDropped.TraceIDZero, 1)
+		return fmt.Errorf("(reason:trace_id_zero): %s", s)
+	}
+	if s.SpanID == 0 {
+		atomic.AddInt64(&ts.TracesDropped.SpanIDZero, 1)
+		return fmt.Errorf("(reason:span_id_zero): %s", s)
```

Comment:
Same here:
```suggestion
		return fmt.Errorf("SpanID is zero: %s", s)
```

Reply:
Same issue as https://github.com/DataDog/datadog-agent/pull/3738#discussion_r298042193, resolving to maintain one discussion.